Firmware/mmu2.cpp

```diff
@@ -861,49 +819,56 @@ StepStatus MMU2::LogicStep(bool reportErrors) {
         }
     }
 
-    if( logic.Running() ){
+    if (logic.Running()) {
         state = xState::Active;
     }
     return ss;
 }
 
 void MMU2::filament_ramming() {
-    execute_extruder_sequence((const E_Step *)ramming_sequence, sizeof(ramming_sequence) / sizeof(E_Step));
+    execute_extruder_sequence(ramming_sequence, sizeof(ramming_sequence) / sizeof(E_Step));
 }
 
 void MMU2::execute_extruder_sequence(const E_Step *sequence, uint8_t steps) {
-    st_synchronize();
+    planner_synchronize();
+    Enable_E0();
+
     const E_Step *step = sequence;
     for (uint8_t i = 0; i < steps; i++) {
-        current_position[E_AXIS] += pgm_read_float(&(step->extrude));
-        plan_buffer_line_curposXYZE(pgm_read_float(&(step->feedRate)));
-        st_synchronize();
+        const float es = pgm_read_float(&(step->extrude));
+        const feedRate_t fr_mm_m = pgm_read_float(&(step->feedRate));
+        planner_set_current_position_E(planner_get_current_position_E() + es);
+        planner_line_to_current_position(MMM_TO_MMS(fr_mm_m));
+
         step++;
     }
+    planner_synchronize(); // it looks like it's better to sync the moves at the end - smoother move (if the sequence is not too long).
```

Comment:
`st_synchronize()` is included with `planner_line_to_current_position()` so I think `planner_synchronize()` doesn't help, only increases code size.